lib/service/Service.ts

```diff
@@ -53,78 +54,20 @@ class Service extends EventEmitter {
 
   /**
    * Get general information about this Exchange Union node.
+   * @throws ServiceError with status.FAILED_PRECONDITION if order constructon fails
    */
   public getInfo = async () => {
     const info: any = {};
 
     info.version = packageJson.version;
-
-    const pairs = await this.orderBook.getPairs();
-    info.numPeers = this.pool.peerCount;
-    info.numPairs = pairs.length;
-
-    let peerOrdersCount: number = 0;
-    let ownOrdersCount: number = 0;
-    for (const key in pairs) {
-      const pair = pairs[key];
-
-      const [peerOrders, ownOrders] = await Promise.all([
-        this.orderBook.getPeerOrders(pair.id, 0),
-        this.orderBook.getOwnOrders(pair.id, 0),
-      ]);
-
-      peerOrdersCount += Object.keys(peerOrders.buyOrders).length + Object.keys(peerOrders.sellOrders).length;
-      ownOrdersCount += Object.keys(ownOrders.buyOrders).length + Object.keys(ownOrders.sellOrders).length;
-    }
-
-    info.orders = {
-      peer: peerOrdersCount,
-      own: ownOrdersCount,
-    };
+    await this.constructOders(info);
```

Comment:
typo in "Oders" - although I don't think we need to break this into a separate method in any case.

Reply:
@sangaman actually this's just for better handling of exceptions